lib/Doctrine/DBAL/Migrations/Finder/RecursiveRegexFinder.php

```diff
@@ -1,60 +1,59 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Doctrine\DBAL\Migrations\Finder;
 
-/**
- * A MigrationFinderInterface implementation that uses a RegexIterator along with a
- * RecursiveDirectoryIterator.
- *
- * @since   1.0.0-alpha3
- */
-final class RecursiveRegexFinder extends AbstractFinder implements MigrationDeepFinderInterface
-{
+use FilesystemIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RegexIterator;
+use const DIRECTORY_SEPARATOR;
+use function sprintf;
 
+final class RecursiveRegexFinder extends Finder implements MigrationDeepFinder
```

Comment:
Can't we replace this whole thing by [RecursiveCallbackFilterIterator](http://php.net/manual/en/class.recursivecallbackfilteriterator.php) + IteratorAggregate + generator?

Reply:
Added issue to investigate this here #626